lambda/test-indexbuild/src/main.ts

```diff
@@ -27,12 +25,38 @@ async function writeIndexData(indexData:RecipeIndex)
   console.log("Done.")
 }
 
-export const handler:Handler = async ()=>{
+export const handler: Handler = async () => {
   console.log("Index test starting up");
 
   console.log("Retrieving index data...");
-  const indexData = await retrieveIndexData();
+  const indexDataForAllRecipes = await retrieveIndexData();
+  const indexDataForUnSponsoredRecipes = {
+    ...indexDataForAllRecipes,
+    recipes: indexDataForAllRecipes.recipes.filter(r=>r.sponsorshipCount===0)
+  }
+  console.log(`Length of unsponsored: ${indexDataForUnSponsoredRecipes.recipes.length}`)
+
+  console.log(`Length of sponsored: ${indexDataForAllRecipes.recipes.length}`);
+
+  for (const entry of indexDataForAllRecipes.recipes) {
+    if(!indexDataForUnSponsoredRecipes.recipes.find(r=>r.capiArticleId===entry.capiArticleId)) {
+      console.log(entry);
+    }
+  }
+
+  const test = indexDataForUnSponsoredRecipes.recipes.find(r=>r.capiArticleId==="lifeandstyle/article/2024/aug/19/testing-sponsor-article-for-v2");
+  console.log("unsponsored: ", test);
+
+  const test2 = indexDataForAllRecipes.recipes.find(r=>r.capiArticleId==="lifeandstyle/article/2024/aug/19/testing-sponsor-article-for-v2");
+  console.log("sponsored: ", test2);
```

Comment:
Can tidy up this debug - I'm happy to do that, I made the mess in the first place 😁